tests/bluetooth/tester/src/audio/btp_pbp.c

```diff
@@ -0,0 +1,267 @@
+/* btp_pbp.c - Bluetooth PBP Tester */
+
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "btp/btp.h"
+
+#include <zephyr/bluetooth/audio/pbp.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+#define LOG_MODULE_NAME bttester_pbp
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_BTTESTER_LOG_LEVEL);
+
+#define PBP_EXT_ADV_METADATA_LEN_MAX 128
+
+static uint8_t pbp_features_cached;
+static uint8_t pbp_metadata_cached_len;
+static uint8_t pbp_metadata_cached[PBP_EXT_ADV_METADATA_LEN_MAX];
+static uint8_t pbp_name_cached_len;
+static uint8_t pbp_broadcast_name_cached[BT_AUDIO_BROADCAST_NAME_LEN_MAX];
+
+static bool scan_get_broadcast_name_len(struct bt_data *data, void *user_data)
+{
+	uint8_t *broadcast_name_len = user_data;
+
+	switch (data->type) {
+	case BT_DATA_BROADCAST_NAME:
+		*broadcast_name_len = data->data_len;
+		return false;
+	default:
+		return true;
+	}
+}
+
+static bool scan_get_data(struct bt_data *data, void *user_data)
+{
+	enum bt_pbp_announcement_feature source_features;
+	uint32_t broadcast_id;
+	uint8_t metadata_len;
+	uint8_t *metadata;
+	struct btp_pbp_ev_public_broadcast_anouncement_found_rp *ev = user_data;
+
+	switch (data->type) {
+	case BT_DATA_BROADCAST_NAME:
+	ev->broadcast_name_len = data->data_len;
+		memcpy(ev->broadcast_name, data->data, data->data_len);
+		return false;
+	case BT_DATA_SVC_DATA16:
+		broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+		sys_put_le24(broadcast_id, ev->broadcast_id);
```

Comment:
You need to verify that it's the correct service data before parsing the broadcast_id: 
```
if (!bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO)) {
```

Similarly, if the service data UUID is `BT_UUID_BROADCAST_AUDIO`, then it is not `BT_UUID_PBA` and `bt_pbp_parse_announcement` will return an error that is not being checked for here. 

So it should be something like
```
if (!bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO)) {
	broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
	sys_put_le24(broadcast_id, ev->broadcast_id);
	return true;
} 

ret = bt_pbp_parse_announcement(data, &source_features, &metadata);
if (ret >= 0) {
	ev->pba_features = source_features;
	return true;
}

return true;
```

Reply:
Ok, added.